src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session.cs

```diff
@@ -64,12 +64,35 @@ internal void UpdateModelTrackingSpan(SnapshotPoint initialPosition)
                 });
             }
 
-            public override void Stop()
+            /// <summary>
+            /// Dismisses the editor completion list.
+            /// 
+            /// NOTE: If calling this, do not call <see cref="CommitEditorSession"/>.
+            /// </summary>
+            public override void DismissEditorSession()
             {
                 AssertIsForeground();
+                this.DisconnectFromPresenter();
+                base.DismissEditorSession();
+            }
+
+            /// <summary>
+            /// Commits and dismisses the underlying editor completion list.
+            /// 
+            /// NOTE: If calling this, do not call <see cref="DismissEditorSession"/>.
+            /// Committing the editor list will implicitly dismiss it.
+            /// </summary>
+            public void CommitEditorSession()
+            {
+                AssertIsForeground();
+                this.DisconnectFromPresenter();
+                this.PresenterSession.Commit();
+            }
+
+            private void DisconnectFromPresenter()
+            {
                 this.PresenterSession.ItemSelected -= OnPresenterSessionItemSelected;
                 this.PresenterSession.ItemCommitted -= OnPresenterSessionItemCommitted;
-                base.Stop();
```

Comment:
This is a weird diff--I added DisconnectFromPresenter to deal with the event unsubscribing, not quite sure where the elided call to base.Stop() came from.

The thrust of this change is that "stopping the computation" is now a separate operation from "dismissing the editor session if it exists". The base Session type used to have a Stop method that did both, but now it has StopComputation, which stops the computation (through CancellationTokens, basically), and DismissEditorSession, which tells the PresenterSession to dismiss.

I should probably make StopComputation sealed, since it's not overridden anywhere. DismissEditorSession is overridden in Completion and SigHelp since they subscribe to slightly different things from their specific presenters.

The base Session.Dismiss unsubscribes from the PresenterSession's Dismiss event and then calls Dismiss--Commit() doesn't call the base dismiss because it doesn't want to call PresenterSession.Dismiss.

Maybe this could be clearer if PresenterSession.Dismiss took a flag indictating whether it's a "commit" dismiss or a "dismiss" dismiss, but then that leaks out into sighelp and quickinfo which have no such concept. Thoughts?